### PR TITLE
Optimize max_pooling for inference for MKL-DNN/IDEEP device

### PR DESCRIPTION
**Pull request by jgong5**
Optimize the max_pooling operation for inference path by setting the "inference" flag to the underlying MKL-DNN, saving the computation and store of max indices which is only needed for training. To make the API compatible, training mode is still the default and inference mode is set in the optimizeForIdeep path.
Test shows the speed-up of a single max_pooling operation is up to 7X on BDW.

### CODE REVIEW

File: caffe2/ideep/operators/pool_op.cc [hunk 1]
```diff
@@ -33,15 +35,18 @@ class IDEEPPoolOp final : public IDEEPConvPoolOpBase {
     auto& X = Input(INPUT);
     auto* Y = Output(OUTPUT);
     auto Y_dims = CalcOutputDims(X, X.get_dim(1));
+    mkldnn::prop_kind pk = training_mode_ ?
+      mkldnn::prop_kind::forward_training : mkldnn::prop_kind::forward_inference;
 
     ideep::pooling_forward::compute(X, Y_dims, *Y,
-        stride_, kernel_, pad_tl(), pad_br(), algo_);
```
Comment (yinghai) [inline caffe2/ideep/operators/pool_op.cc]: Just curious, what's the previous default that was causing the slowdown?

Comment (jgong5) [reply caffe2/ideep/operators/pool_op.cc]: @yinghai The default setting is training...

Comment (yinghai) [reply caffe2/ideep/operators/pool_op.cc]: OK...

**Comment by jgong5** [comment]
@yinghai OK to merge?

**Pull request closed without merging**